### ISSUE TITLE
New MacOS ABI darwin-wx315

### ISSUE THREAD

**Issue by leamas** [Rasbats/shipdriver_pi#209]
As heading says, and according to email thread. 5.6.0 introduces an ABI break, the new ABI uses wxWidgets 3.1.5 and is named _darwin-wx315_.

See also https://github.com/OpenCPN/plugins/pull/452

Part  of this is also that it is decided that MacOS plugins should support 10.9, which means new package handling and longer build times. The fixes are available in main OpenCPN.

This will be some work. It will mean that the current darwin plugins are sunset, there will be no updates available for them, all updates will be for the new ABI.

**Comment by Rasbats** [comment]
Saw this from the introduction of the 5.5.920 beta. I can make betas for MacOS and prune out the other betas that appear in auto so the beta catalog is not polluted when the squash with master takes place. Could you look at the changes needed for the `circleci-build-macos.sh` script?

**Comment by leamas** [comment]
Probably, but ENOTIME right now... perhaps end of week (?)

**Comment by Rasbats** [comment]
Not sure if it helps ShipDriver but I needed a MacOS wx315 for JavaScript_pi.
The work here:
[https://github.com/bdbcat/oesenc_pi](https://github.com/bdbcat/oesenc_pi) was raided for the cmake directory and the ci files  _circleci-build-macos.sh_  and _upload.sh.in_. 
MacOS and Linux builds seem Ok. Flatpak A64 and Android fail to build. Logs are here:
[https://app.circleci.com/pipelines/github/Rasbats/JavaScript_pi](https://app.circleci.com/pipelines/github/Rasbats/JavaScript_pi)

**Comment by Rasbats** [comment]
ShipDriver has a successful alpha build. Flatpak A64 had a ca certificates problem, resolved by updating the certificates in the _circleci-build-flatpak.sh_ script.
The MacOS is using wxWidgets 3.1.5. With 3.1.2 the cache could not be reloaded. Have not tried reverting to 3.1.2.
Android is building ok, I think.

Now building again for a bug causing problems for OpenCPN: [https://github.com/OpenCPN/OpenCPN/issues/2418#issuecomment-939203439](https://github.com/OpenCPN/OpenCPN/issues/2418#issuecomment-939203439 )

Alec: Can you see any problems with my changes?

**Comment by leamas** [comment]
I will try to give shipdriver some love tomorrow, time permitting. A bit too much these days, but I think I'm done with the OpenCPN work which is prioritized during the beta window.

**Comment by rgleason** [comment]
Is it ok to use it now for Logbook, or should I wait?

**Comment by Rasbats** [comment]
>I. e., OpenCPN/OpenCPN#2418.

Not quite. That issue related to a crash of OpenCPN. The issue raised on cruisersforum was the display of a jigsaw icon if the steps shown in the post were followed. Dave asked for issues with the beta to be added to the forum posts. Should I raise an issue on GitHub as well?

**Comment by Rasbats** [comment]
Latest log for MacOS build:
[https://app.circleci.com/pipelines/github/Rasbats/shipdriver_pi/556/workflows/b8e5b95d-03ab-43e9-84f4-bc4f015252f6/jobs/4017](https://app.circleci.com/pipelines/github/Rasbats/shipdriver_pi/556/workflows/b8e5b95d-03ab-43e9-84f4-bc4f015252f6/jobs/4017)

**Comment by leamas** [comment]
> Latest log for MacOS build:

OK, I can reproduce. Late now, tomorrow I need to lift a sailing yacht out of some water that soon will become ice...

**Comment by leamas** [comment]
The broken macos build is about the cache. You could walk around it for now by not using any cache at all according to patch below. But such builds will take a long time, I will look into how t handle the cache problem in a more permanent way later.

```
diff --git a/.circleci/config.yml b/.circleci/config.yml
index 6f95bf9..f0c69c7 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -67,18 +67,7 @@ jobs:
       - OCPN_TARGET: macos
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - macos-cache-v1-{{checksum "ci/circleci-build-macos.sh"}}
       - run: ci/circleci-build-macos.sh
-      - save_cache:
-          key: macos-cache-v1-{{checksum "ci/circleci-build-macos.sh"}}
-          paths:
-            - /usr/local/Homebrew
-            - /usr/local/Caskroom
-            - /usr/local/Cellar
-            - /usr/local/lib
-            - /usr/local/include
       - run: sh -c "otool -L build/app/*/OpenCPN.app/Contents/PlugIns/*.dylib"
       - run: cd build; /bin/bash < upload.sh
       - run: python3 ci/git-push

```

**Comment by rgleason** [comment]
Mike, Alec  Logbook has 2 android 3 flatpak and 1 macos failing.
https://github.com/rgleason/LogbookKonni_pi/tree/update-sd

Is there going to be an update of the template?
Or should I just push what I have?

**Comment by leamas** [comment]
There will be an update, unclear when. The best is probably to wait.

**Comment by leamas** [comment]
> MacOS and Linux builds seem Ok. Flatpak A64 and Android fail to build. Logs are here:

The flatpak error should be fixed by a13a6c18.

The Android errors seems specific for the javascript plugin, missing includes. You also need to disable some warnings, the sheer amount of warnings makes it too hard to see the errors.

**Comment by leamas** [comment]
@rasbats^

**Comment by leamas** [comment]
The move to darwin-wx315 is actually just a one-line patch in the cmake directory, the rest of the changes in oesenc represents older versions used there.

The real work is about the included libraries. The brew libraries used (libexif, perhaps gettext) are not built for the ancient 10.9. So, even if the build is OK the resulting plugin is not usable on a 10.9 system.

Dave worked around this by using another  packaging system instead of brew. I need to refresh the details and incorporate these changes.

**Comment by leamas** [comment]
Fixed in 1c05fe28. Closing
